proposals/partial-type-inference-v2.md

```diff
@@ -0,0 +1,398 @@
+- If `F` is generic and `M` has no type argument list, `F` is a candidate when:
+  - Type inference (§12.6.3) succeeds, inferring a list of type arguments for the call, and
+  - Once the inferred type arguments are substituted for the corresponding method type parameters, all constructed types in the parameter list of `F` satisfy their constraints ([§8.4.5](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/types.md#845-satisfying-constraints)), and the parameter list of `F` is applicable with respect to `A` (§12.6.4.2)
+- **If `F` is generic and `M` is *partial_inferred*, `F` is a candidate when:**
+  - **Method type inference (See the [Type Inference] section) succeeds, inferring the type arguments list for the call, and**
+  - **Once the inferred type arguments are substituted for the corresponding method type parameters, all constructed types in the parameter list of `F` satisfy their constraints ([§8.4.5](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/types.md#845-satisfying-constraints)), and the parameter list of `F` is applicable with respect to `A` ([§12.6.4.2](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#12642-applicable-function-member))**
+- If `F` is generic and `M` includes a type argument list, `F` is a candidate when:
+  - `F` has the same number of method type parameters as were supplied in the type argument list, and
+  - Once the type arguments are substituted for the corresponding method type parameters, all constructed types in the parameter list of `F` satisfy their constraints (§8.4.5), and the parameter list of `F` is applicable with respect to `A` (§12.6.4.2).
+
+### Object creation expressions
+
+The binding-time processing of an [*object_creation_expression*](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#128162-object-creation-expressions) of the form new `T(A)`, where `T` is a *class_type*, or a *value_type*, and `A` is an optional *argument_list*, is changed in the following way.
+
+The binding-time processing of an *object_creation_expression* of the form new `T(A)`, where `T` is a *class_type*, or a *value_type*, and `A` is an optional *argument_list*, consists of the following steps:
+
+- If `T` is a *value_type* and `A` is not present:
+  - **The *object_creation_expression* is a default constructor invocation.**
+    - **If the type is *partial_inferred*, Constructor type inference (See the [Type Inference] section) of the default constructor occurs to determine closed type. If it succeeded, construct the type using the inferred type arguments. If it failed the binding-time error occurs.**
+    - **If the type inference above succeeded or the type is not inferred, the result of the *object_creation_expression* is a value of (constructed) type `T`, namely the default value for `T` as defined in [§8.3.3](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/types.md#833-default-constructors).**
```

Comment:
I'm conflicted about having these lines here because:
- they are out of date (`new ValueType()` can be invocation of a parameterless (explicitly declared) struct constructor).
- the described inference will never succeed. Perhaps in future if target type contributes to inference, but not with this version of the feature.

I think for the purposes of this feature speclet we could delete these lines and just remove "Otherwise," from the "Otherwise, if `T` is a *type parameter*"... below.